lib/zhr_devs/identity_management/identity.ex

```diff
@@ -10,52 +10,41 @@ defmodule ZhrDevs.IdentityManagement.Identity do
   alias Uptight.Result
   alias Uptight.Text, as: T
 
-  import Uptight.Assertions
-
   defdata do
     identity :: Uptight.Text.t()
     hashed_identity :: Uptight.Base.Urlsafe.t()
     login_at :: UtcDateTime.t()
   end
 
-  alias DomaOAuth.Authentication.Success
+  alias ZhrDevs.IdentityManagement.Events.LoggedIn
 
-  # Public API
+  alias EventStore.RecordedEvent
 
-  def start_link(%Success{hashed_identity: hashed_identity} = success_struct) do
-    GenServer.start_link(__MODULE__, success_struct, name: via_tuple(hashed_identity))
-  end
+  alias ZhrDevs.IdentityManagement.Identity
+
+  # Public API
 
-  @spec renew_login(binary()) :: :ok | no_return()
-  def renew_login(hashed_identity) when is_binary(hashed_identity) do
-    GenServer.call(via_tuple(hashed_identity), :renew_login)
+  def start_link(%LoggedIn{hashed_identity: hashed_identity} = event) do
+    GenServer.start_link(__MODULE__, event, name: via_tuple(hashed_identity.encoded))
   end
 
   # Callbacks
 
   @impl GenServer
-  def init(success) do
-    parsed_success = success_to_identity(success)
-    assert Result.is_ok?(parsed_success), "Parsing success struct is failed"
+  def init(event) do
+    :ok = ZhrDevs.EventStore.subscribe(event.hashed_identity.encoded)
```

Comment:
Uh oh, I don't understand what does it mean to subscribe here. Let's say we have an event "sign out", could we then handle it in `handle_info` to, say shutdown?